runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -363,9 +363,12 @@ def _start_ray_workers(self, ray_port, env):
         # Find the internal IP corresponding to the public_head_ip and the rest are workers
         internal_head_ip = None
         worker_ips = []
-        stable_internal_external_ips = self._sky_status()[
-            "handle"
-        ].stable_internal_external_ips
+        sky_status = self._sky_status()
```

Comment:
wondering if we can get rid of all this and just do `for internal, external in self.stable_internal_external_ips` below, since it should already be set by the time `_start_ray_workers` is called in restart server.

not sure if there's any edge case though so fine to keep this for the release and test later on if we can skip this extra sky status/populating